ome_zarr/reader.py

```diff
@@ -177,6 +204,41 @@ def __init__(self, node: Node) -> None:
     def lookup(self, key: str, default: Any) -> Any:
         return self.zarr.root_attrs.get(key, default)
 
+    def get_schema(self, strict: bool = False) -> Optional[Dict]:
+        # If not implemented then validate will be no-op
+        return None
+
+    def validate(self, strict: bool = False) -> None:
+        # multiscales = self.lookup("multiscales", [])
+        # version = multiscales[0].get("version", CurrentFormat().version)
+        schema = self.get_schema()
+        if schema is None:
+            LOGGER.info("No schema for %s" % self.zarr)
+            return
+        LOGGER.info("Validating Multiscales spec at: %s" % self.zarr)
+
+        # Always do a validation with the MUST rules
+        # Will throw ValidationException if it fails
+        json_data = self.zarr.root_attrs
+        jsonschema_validate(instance=json_data, schema=schema)
+
+        # If we're also checking for SHOULD rules,
+        # we want to iterate all errors and show as Warnings
+        if strict:
+            strict_schema = self.get_schema(strict=True)
+            if strict_schema is None:
+                return
+            # we only need this store to allow use of cached schemas
+            # (and potential off-line use)
+            schema_store = {
+                schema["$id"]: schema,
+                strict_schema["$id"]: strict_schema,
+            }
+            resolver = RefResolver.from_schema(strict_schema, store=schema_store)
+            validator = Validator(strict_schema, resolver=resolver)
+            for error in validator.iter_errors(json_data):
+                LOGGER.warn(error.message)
```

Comment:
As in https://github.com/ome/ome-zarr-py/pull/142#discussion_r905076199 above, I'm thinking of separating the reporting of errors from the validation itself. So, the cli `validate` command can log or print errors, but the `Spec.validate()` and `utils.validate()` would only *return* errors / messages (and maybe log at INFO level)?

It could get tricky to give the user an accurate picture of what failed: eg. when you validate an Image with Labels, the `image` schema is used for both, so you can get an output like this and it's hard to know which object was missing "name":

```
$ ome_zarr validate 6001247.zarr/ --strict --clear_cache
WARNING:ome_zarr.reader:'metadata' is a required property
WARNING:ome_zarr.reader:'type' is a required property
WARNING:ome_zarr.reader:'name' is a required property
WARNING:ome_zarr.reader:'metadata' is a required property
WARNING:ome_zarr.reader:'type' is a required property
```

Reply:
In the spirit of `jsonschema.validate` vs `jsonschema.iter_errors`, I concur with the idea of having an API that returns an iterable of all exceptions.

Will the contract be the same for the default and the strict schemas? In the current implementation, the base schema raises an exception but the strict schema does not.
Also I assume since the strict schema extends the base schema, it is not necessary to have 2 validation calls?